### PR TITLE
Merge incomplete Linux build system work

### PR DESCRIPTION
**Pull request by tbodt**
This has been in a branch so far, want to merge it in now so I can keep working on it without dealing with branches. Shouldn't affect the non-Linux configuration at all.

### CODE REVIEW

File: app/TerminalViewController.m [hunk 1]
```diff
@@ -197,11 +205,17 @@ - (void)processExited:(NSNotification *)notif {
             return;
         }
     }
-#if !ISH_LINUX
     current = NULL; // it's been freed
-#endif
     [self startNewSession];
+    [UIAlertController alertControllerWithTitle:nil message:nil preferredStyle:nil];
```
Comment (saagarjha) [inline app/TerminalViewController.m:212]: :/

**Review by saagarjha — CHANGES_REQUESTED** [review]
Normal build appears to work.

File: app/TerminalViewController.m (hunk 1 above)
Comment (tbodt) [reply app/TerminalViewController.m:212]: yeah what is this

**Review by saagarjha — APPROVED** [review]
lgtm aside from your commit messages

**Pull request merged**